src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx

```diff
@@ -44,35 +56,28 @@ function QRCodeScannerScreen() {
 
   const handleBarCodeScanned = async (result: BarCodeScannerResult) => {
     const caseId = result.data;
-    if (!scanned) {
-      setScanned(true);
-      const valid = await isValidCase(caseId);
-      if (!valid) {
-        // TODO: Display error toast message
-        setToast('Not a valid QRCODE!');
-        setScanned(false); // TODO: setTimeout here
-        return;
-      }
-      const duplicate = await containsDuplicateCase(caseId);
-      if (duplicate) {
-        setToast('DUPLICATES NOT ALLOWED!');
-        setScanned(false); // TODO: setTimeout here as well
-        return;
-      }
-      const caseData: Case = await getCaseById(caseId);
-      const { id, title, imageUrl, date, lawFirm, summary } = caseData;
-      router.push({
-        pathname: '/AllCases/QRCodeScanner/AddCase',
-        params: {
-          id,
-          title,
-          imageUrl,
-          date,
-          lawFirm,
-          summary,
-        },
-      });
+    if (!validIds.includes(caseId)) {
+      // TODO: Display error toast message
+      setToast('Not a valid QRCODE!');
+      return;
+    }
+    if (userIds.includes(caseId)) {
+      setToast('DUPLICATES NOT ALLOWED!');
+      return;
```

Comment:
We need to check `scanned` here still for valid cases because letting a user scan infinitely in a valid case will lead to the same issue of spamming back and forth between the QR Code Scanner screen and Add Case screen if the user clicks cancel. I think it makes most sense to do a `if-else if` clause here -- if (!validIds.includes(caseId)), else if  (userIds.includes(caseId)), else if (!scanned). Let me know if this makes sense!